### PR TITLE
[PropertyInfo] Fix ReflectionExtractor + minor tweaks

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | 5.1 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix N/A 
| License       | MIT
| Doc PR        | N/A

Spotted while rebasing #32133

### CODE REVIEW

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php [hunk 1]
```diff
@@ -308,16 +308,16 @@ public function getWriteInfo(string $class, string $property, array $context = [
             $mutator->setRemoverInfo(new PropertyWriteInfo(PropertyWriteInfo::TYPE_METHOD, $removerAccessName, $this->getWriteVisiblityForMethod($removerMethod), $removerMethod->isStatic()));
 
             return $mutator;
-        } else {
-            $errors = array_merge($errors, $adderAndRemoverErrors);
         }
 
+        array_push($errors, ...$adderAndRemoverErrors);
```
Comment (ogizanagi) [inline src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:313]: Mostly motivated by the usage of array_merge in loops, which are more greedy for memory consumption, but updated everywhere so it's consistent, and we don't really need a merge anyway (keys don't matter)

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php [hunk 2]
```diff
@@ -348,25 +350,27 @@ public function getWriteInfo(string $class, string $property, array $context = [
         [$accessible, $methodAccessibleErrors] = $this->isMethodAccessible($reflClass, '__set', 2);
         if ($accessible) {
             return new PropertyWriteInfo(PropertyWriteInfo::TYPE_PROPERTY, $property, PropertyWriteInfo::VISIBILITY_PUBLIC, false);
-        } else {
-            $errors = array_merge($errors, $methodAccessibleErrors);
         }
 
-        [$accessible, $methodAccessibleErrors] = $this->isMethodAccessible($reflClass, '__call', 2);
```
Comment (ogizanagi) [inline src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:355]: No need to compute method is accessible nor adding errors if the magic `__call` isn't allowed, right?

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php (hunk 1 above)
Comment (ogizanagi) [reply src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php]: Unfortunately, in php <7.3, if `$adderAndRemoverErrors` is empty: 

```
array_push() expects at least 2 parameters, 1 given
```

**Comment by fabpot** [comment]
Thank you @ogizanagi.

**Pull request merged**